ui/app/components/feedback/HumanFeedbackModal.tsx

```diff
@@ -3,218 +3,31 @@ import {
   DialogContent,
   DialogHeader,
   DialogTitle,
+  DialogTrigger,
 } from "~/components/ui/dialog";
-import { useConfig } from "~/context/config";
-import MetricSelector from "../metric/MetricSelector";
-import { useState } from "react";
-import type { ContentBlockOutput } from "~/utils/clickhouse/common";
-import type { JsonInferenceOutput } from "~/utils/clickhouse/common";
-import Output from "../inference/Output";
-import { Link, Form } from "react-router";
-import { Button } from "~/components/ui/button";
-import {
-  filterMetricsByLevel,
-  filterStaticEvaluationMetrics,
-} from "~/utils/config/metric";
-import BooleanFeedbackInput from "./BooleanFeedbackInput";
-import FloatFeedbackInput from "./FloatFeedbackInput";
-import CommentFeedbackInput from "./CommentFeedbackInput";
 
 interface HumanFeedbackModalProps {
   isOpen: boolean;
-  onClose: () => void;
-  // This should be provided if the feedback is for an inference
-  // and omitted if the feedback is for an episode
-  inferenceOutput?: ContentBlockOutput[] | JsonInferenceOutput;
-  // Exactly one of the following should be provided
-  episodeId?: string;
-  inferenceId?: string;
+  onOpenChange: (isOpen: boolean) => void;
+  trigger?: React.ReactElement;
+  children?: React.ReactNode;
 }
 
 export function HumanFeedbackModal({
   isOpen,
-  onClose,
-  inferenceOutput,
-  episodeId,
-  inferenceId,
+  onOpenChange,
+  trigger,
+  children,
 }: HumanFeedbackModalProps) {
-  // Exactly one of episodeId and inferenceId should be provided
-  if ((episodeId && inferenceId) || (!episodeId && !inferenceId)) {
-    throw new Error(
-      "Exactly one of episodeId and inferenceId should be provided. This is a bug. Please file a bug report at https://github.com/tensorzero/tensorzero/discussions/new?category=bug-reports",
-    );
-  }
   return (
-    <Dialog open={isOpen} onOpenChange={onClose}>
+    <Dialog open={isOpen} onOpenChange={onOpenChange}>
+      {!!trigger && <DialogTrigger asChild>{trigger}</DialogTrigger>}
```

Comment:
Generally speaking I'd recommend using the dialog trigger component instead of using `onClick` to connect dialogs to triggers (there are a few usability tweaks under the hood in Radix, plus you just don't need to manage as much state in many cases).

Reply:
I didn't know it existed!